### ISSUE TITLE
Why the teacher model is not with DDP when using vit, and the teacher weight is updated by teacher_without_ddp instead of  teacher.module?

### ISSUE THREAD

**Issue by pansanity666** [facebookresearch/dino#157]
Hi, 

I have two questions about the DDP:

1. In the code, you move the teacher model to DDP only if there is batch normalization (i.e., resnet50 ?). Why don't you move teacher to DDP when there is no BN in the model (i.e., vit)? 
2. I find that you use the```teacher_without_ddp``` to update the teacher's parameters. Can I directly use teacher.module in the case of resnet50? If not, what's the difference? Does it relate to the  mechanism of DDP?

Best,

**Comment by mathildecaron31** [comment]
Hi @pansanity666 

1. I noticed that when using DDP the model uses a bit more GPU memory (maybe that is not the case with newest pytorch I don't know) that is why I preferred not to use it when not needed.

2. Yes I think you can use directly teacher.module !

**Comment by yfchen97** [comment]
Hi,
If i run codes in single GPU, can i use teacher to replace all teacher_without_ddp?
Best,
